src/shims/unix/fs.rs

```diff
@@ -395,7 +395,9 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         // Isolation check is done via `FileDescriptor` trait.
 
         let seek_from = if whence == this.eval_libc_i32("SEEK_SET") {
-            SeekFrom::Start(u64::try_from(offset).unwrap())
+            // Be consistent with standard library.
+            #[allow(clippy::cast_sign_loss)]
+            SeekFrom::Start(i64::try_from(offset).unwrap() as u64)
```

Comment:
We're implementing POSIX lseek here, it doesn't matter what the Rust standard library does.

The real question is what does POSIX say should happen when `offset` is negative in SEEK_SET mode?